lib/unleash/feature_toggle.rb

```diff
@@ -40,8 +40,10 @@ def get_variant(context, fallback_variant = disabled_variant)
       return disabled_variant unless self.enabled && am_enabled?(context, true)
       return disabled_variant if sum_variant_defs_weights <= 0
 
+      stickiness = self.variant_definitions[0].stickiness || "default"
```

Comment:
The `[0]` makes me uncomfortable. 

Why should it take the first variant in the list? What happens if you try to `get_variant()` from a feature toggle with no variants? (`self.variant_definitions[0]` would be `nil` and `nil.stickiness` would throw a `NoMethodError`).

Reply:
So we've included the stickiness in each variant in the list for backwards compatibility reasons in the variant list

variant_definitions _should_ never be nil but I realize these are famous last words and your comment is entirely fair. Let me take another look at this today and see if I can structure this in a way that doesn't looks so dangerous

Reply:
I want to say this should be safer now - code should try find the first non nil variant or default to "default"